fipy/tools/numerix.py

```diff
@@ -1007,6 +1007,36 @@ def in1d(ar1, ar2, assume_unique=False):
         else:
             return flag[indx][rev_idx]
 
+# Buggy numpy.ma.in1d fixed in https://github.com/numpy/numpy/pull/20011
+def in1dMA(ar1, ar2, assume_unique=False, invert=False):
```

Comment:
Do you want to add the argument description? Would this function work if ar1 is not masked but ar2 is masked?

Reply:
Documentation was taken directly from `numpy.ma.in1d`, but I've now lifted the documentation from `numpy.in1d`.

The first thing it does is convert `ar1` and `ar2` to masked arrays, so it should work for any.